src/mpl/configure.ac

```diff
@@ -154,10 +154,23 @@ AC_ARG_ENABLE(fast,
         none     - Disable alwaysinline (default)
 ],,enable_fast=none)
 
-# enable-fast
-# only alwaysinline option is valid but still need break down multiple options
+# enable-fast can have multiple options, need break down options
 # transferred from MPICH separated by commas
 enable_fast_alwaysinline=no
+
+# Instruction specific options (sse2, avx, avx512f) are for users to explicitly request for
+# corresponding instruction sets.
+# If these options are not set. We will fallback to compiler-time and runtime-check, and enable
+# the instructions that is supported by the compiler and the building machine.
+#
+# If set, we only check if the compiler supports them and skip
+# the runtime check. For example, on AMD Zen3 with avx512f option set, MPICH will be built with
+# AVX512F support as compiler generally supports it. But the library is broken for Zen3 because
+# the CPU don't have AVX512F support.
```

Comment:
Looking at this commit on its own, this comment is not accurate. Neither is the commit message "...through both compile-time and runtime checks". This code only uses runtime checks.